src/query/src/log_query/planner.rs

```diff
@@ -371,12 +424,29 @@ impl LogQueryPlanner {
                 // nothing to do, return empty vec.
             }
             LogExpr::Alias { expr, alias } => {
-                let expr = self.log_expr_to_column_expr(expr, plan_builder.schema().as_arrow())?;
-                let aliased_expr = expr.alias(alias);
+                let schema = plan_builder.schema();
+                let df_expr = self.log_expr_to_df_expr(expr, schema)?;
+                let aliased_expr = df_expr.alias(alias);
                 plan_builder = plan_builder
                     .project([aliased_expr.clone()])
                     .context(DataFusionPlanningSnafu)?;
             }
+            LogExpr::BinaryOp { left, op, right } => {
+                let schema = plan_builder.schema();
+                let left_expr = self.log_expr_to_df_expr(left, schema)?;
+                let right_expr = self.log_expr_to_df_expr(right, schema)?;
+                let df_op = Self::binary_operator_to_df_operator(op)?;
+
+                let binary_expr = Expr::BinaryExpr(BinaryExpr {
+                    left: Box::new(left_expr),
+                    op: df_op,
+                    right: Box::new(right_expr),
+                });
```

Comment:
non-blocking: Looks like we have to repeat similar logic in log_expr_to_df_expr.